frontend/src/app/Settings/Facility/FacilityForm.tsx

```diff
@@ -226,16 +181,20 @@ const FacilityForm: React.FC<Props> = (props) => {
   };
 
   const validateFacilityAddresses = async () => {
-    const originalFacilityAddress = getFacilityAddress(facility);
-
+    let updatedValues = getUpdatedValues();
+    const originalFacilityAddress = getFacilityAddress(updatedValues.facility);
     const zipCodeData = await getZipCodeData(originalFacilityAddress.zipCode);
     const isValidZipForState = isValidZipCodeForState(
       originalFacilityAddress.state,
       zipCodeData
     );
 
     if (!isValidZipForState) {
-      showError("Invalid ZIP code for this state", "Form Errors");
+      setError("facility.zipCode", {
```

Comment:
⚠️ currently only validating if the facility's zip code and state combination is valid

we are not checking if the ordering provider's zip code and state (if provided, since it is not required) are valid combinations -- this is the current behavior as well